doc/refman/Classes.tex

```diff
@@ -400,18 +444,105 @@ \subsection{\tt Typeclasses Transparent, Opaque {\ident$_1$ \ldots \ident$_n$}}
 
 This is equivalent to {\tt Hint Transparent,Opaque} {\ident} {\tt: typeclass\_instances}.
 
+\subsection{\tt Set Typeclasses Module Eta}
+\optindex{Typeclasses Modulo Eta}
+
+This allows eta-conversion during unification of type-classes.
+
+\subsection{\tt Set Typeclasses Dependency Order}
+\optindex{Typeclasses Dependency Order}
+
+This option (now on by default) respects the dependency order between
+subgoals, meaning that subgoals which are depended on by other subgoals 
+come first, while the non-dependent subgoals were put before the
+dependent ones previously (Coq v8.5 and below). This can result in quite
+different performance behaviors of proof search.
+
+\subsection{\tt Set Typeclasses Legacy Resolution}
+\optindex{Typeclasses Legacy Resolution}
+
+This option (off by default) uses the 8.5 implementation of resolution.
+Use for compatibility purposes only (porting and debugging).
+
+\subsection{\tt Set Typeclasses Limit Intros}
+\optindex{Typeclasses Limit Intros}
+
+This option (on by default in Coq 8.6 and below), controls the ability
+to apply hints while avoiding eta-expansions in the proof term
+generated. It does so by allowing hints that conclude in an product to
+apply to a goal with a matching product directly, avoiding an
+introduction. \emph{Warning:} can be expensive as it requires rebuilding
+hint clauses dynamically, and does not benefit from the invertibility
+status of the product introduction rule, resulting in more expensive
+proof-search (i.e. more useless backtracking).
+
+\subsection{\tt Set Typeclasses Filtered Unification}
+\optindex{Typeclasses Filtered Unification}
+
+This option, available since Coq 8.6, switches the hint application
+procedure to a filter-then-unify strategy. To apply a hint, we first
+check that it \emph{matches} syntactically the inferred or specified
+pattern of the hint, and only then try to \emph{unify} the goal with the
+conclusion of the hint. This can drastically improve performance by
+calling unification less often, matching syntactic patterns being a much
+quicker operation than unification. This also provides more control on
+the triggering of instances. For example, forcing a constant to
+explicitely appear in the pattern will make the hint never apply on a
+goal where there is a hole in that place.
+
+\subsection{\tt Set Typeclass Resolution After Apply}
+\optindex{Typeclasses Resolution After Apply}
+\emph{Deprecated since 8.6}
+
+This option (off by default in Coq 8.6 and 8.5) controls the resolution
+of typeclass subgoals generated by the {\tt apply} tactic.
+
+
+\subsection{\tt Set Typeclass Resolution For Conversion}
+\optindex{Typeclasses Resolution For Conversion}
+
+This option (on by default) controls the use of typeclass resolution
+when a unification problem cannot be solved during elaboration. 
+With this option on, typeclasses resolution is tried in this case before
```

Comment:
The use of "before " makes it sound like the alternative is "after" rather than "not at all".  I'd replace "this case before" with "this case, after which"